.github/workflows/ci.yml

```diff
@@ -17,17 +17,22 @@ env:
 
 jobs:
   clang-format:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          ref:  ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.9' 
       - name: Get clang-format
-        run: sudo apt-get install -yqq clang-format-6.0
+        run: sudo apt-get install -yqq clang-format
       - name: Applying clang-format for changed files
         run: |
-          FILES=$(git diff --diff-filter=d --name-only ${{ github.event.pull_request.base.sha }} | grep ^include | grep -v nanorange\.hpp\$ || true)
-          echo $FILES | xargs -n1 -t -r clang-format-6.0 --style=file -i
+          FILES=$(git diff --diff-filter=d --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | grep ^include | grep -v nanorange\.hpp\$ || true)
+          CLANG_FORMAT_DIFF_PATH=$(which clang-format-diff)
+          echo $FILES | xargs -n1 -t -r git diff -U0 --no-color --relative ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | python3 $CLANG_FORMAT_DIFF_PATH -i -p1 -style file
```

Comment:
see if we can simplify these lines because we have pull request head checked out from line 25.

Reply:
These lines were simplified by removing reference to the current working copy commit (as it is implied), and abstracting out the merge base commit to a variable.